From 314b07fb245248136f9e37fe76559a8d68db8c92 Mon Sep 17 00:00:00 2001
From: Helen Huang <he.huang@intel.com>
Date: Mon, 19 Apr 2021 16:06:15 +0800
Subject: [PATCH] Add logs to fwVersionIsSame()

Add logs to fwVersionIsSame() to indicate whether the firmware
version is changed or not.

Tested:
Logs are printed as expected when firmware updating and BMC rebooting.

Log of rebooting:
The firmware version is similiar as the last boot,
Hash value of versionFile is:3336889560

Log of Firmware update:
The firmware version is changed since the last boot,
hash value of current versionFile is:3336889560,
hash value of versionFile of last boot is:834871226

Change-Id: I5306917329d2e2e015af58cad1e9c59881f0b217
Signed-off-by: Helen Huang <he.huang@intel.com>
---
 src/utils.hpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/utils.hpp b/src/utils.hpp
index 99e2f85..2ae1381 100644
--- a/src/utils.hpp
+++ b/src/utils.hpp
@@ -121,13 +121,22 @@ inline bool fwVersionIsSame(void)
 
         if (expectedHash == hashString)
         {
+            std::cout << "The firmware version is similiar as the last boot, "
+                         "hash value of versionFile is:"
+                      << hashString.c_str() << "\n";
             return true;
         }
+        std::cout << "The firmware version is changed since the last boot, hash "
+                     "value of current versionFile is:"
+                  << expectedHash.c_str()
+                  << ", hash value of versionFile of last boot is:"
+                  << hashString.c_str() << "\n";
         hashFile.close();
     }
 
     std::ofstream output(versionHashFile);
     output << expectedHash;
+
     return false;
 }
 
-- 
2.25.1

